### DISCUSSION TITLE
KoDefaultValueProviderCore does not handle default values that are "object"s

### DISCUSSION THREAD

**Discussion by drewhannay** [LemonAppDev/konsist#1269] [Report a Problem] [answered]
### Tell us what version of Konsist you use.

0.15.1

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

This Konsist test is a simplified sample that just asserts that function parameters have default values:
```kotlin
@Test
fun `default parameter objects`() {
    Konsist.scopeFromProduction()
        .functions()
        .assertTrue { function ->
            function.parameters.all { parameter ->
                parameter.defaultValue != null
            }
        }
}
```

Here is a sample function which SHOULD pass the test but does not:
```kotlin
object Bar

fun foo(
    foo: Bar = Bar,
) = "Foo"
```

### Actual behaviour

Any function parameters whose default values are an `object` are not recognized by `KoDefaultValueProviderCore`. It returns `null` as if they do not have any default value.

### Expected behaviour

`KoDefaultValueProviderCore` should handle this `object` case and return the string text of the `object`. Right now it only handles primitive values and function calls. This is especially relevant in code like Jetpack Compose where a `Modifier` object is often used as a default value

**Comment by nataliapeterwas** [comment, marked as answer]
Thank you for reporting the problem.

We will fix this in the next release: [PR](https://github.com/LemonAppDev/konsist/pull/1282) :)

**Comment by nataliapeterwas** [comment]
Konsist `0.16.0` has been released! This release contains a fix for this issue :)
